subresource-integrity/subresource-integrity.html

```diff
@@ -6,74 +6,176 @@
-
-  test(function() {
-    assert_true(loads_scripts_with_base64_encoded_sha_digests)
-  }, "Loads scripts with base64 encoded sha digests");
+  var xorigin_anon = location.protocol
+    + '//www1.' + location.hostname + ':' + location.port
+    + '/subresource-integrity/crossorigin-anon.js';
+
+  var xorigin_creds = location.protocol
+    + '//www1.' + location.hostname + ':' + location.port
+    + '/subresource-integrity/crossorigin-creds.js';
+
+  var xorigin_ineligible = location.protocol
+    + '//www1.' + location.hostname + ':' + location.port
+    + '/subresource-integrity/crossorigin-ineligible.js';
+
+    var SRIScriptTest = function(pass, name, src, integrityValue, crossoriginValue) {
+        this.pass = pass;
+        this.name = name;
+        this.src = src;
+        this.integrityValue = integrityValue;
+        this.crossoriginValue = crossoriginValue;
+    }
+
+    SRIScriptTest.prototype.execute = function() {
+        var test = async_test(this.name);
+        var e = document.createElement("script");
+        e.src = this.src;
+        e.setAttribute("integrity", this.integrityValue);
+        if(this.crossoriginValue) {
+            e.setAttribute("crossorigin", this.crossoriginValue);
+        }
+        if(this.pass) {
+            e.addEventListener("load", function() {test.done()});
+            e.addEventListener("error", function() {
+                test.step(function(){ assert_unreached("Good load fired error handler.") })
+            });
+        } else {
+           e.addEventListener("load", function() {
+                test.step(function() { assert_unreached("Bad load succeeded.") })
+            });
+           e.addEventListener("error", function() {test.done()});
+        }
+        document.body.appendChild(e);
+    };
+
+    new SRIScriptTest(
+        true,
+        "Same-origin script with correct hash.",
+        "matching-digest.js",
+        "sha256-EKclUXH9SRRUv3FmL7bIEV0z2s3EvzHFxzHKCnfHT/E="
+    ).execute();
+
+    new SRIScriptTest(
+        false,
+        "Same-origin script with incorrect hash.",
+        "non-matching-digest.js",
+        "sha256-deadbeefdeadbeefdeadbeefdeadbeefdeadbeefdead"
+    ).execute();
+
+    new SRIScriptTest(
+        true,
+        "Same-origin script with multiple sha256 hashes, including correct.",
+        "matching-digest.js",
+        "sha256-EKclUXH9SRRUv3FmL7bIEV0z2s3EvzHFxzHKCnfHT/E= sha256-deadbeefdeadbeefdeadbeefdeadbeefdeadbeefdead"
+    ).execute();
+
+    new SRIScriptTest(
+        true,
+        "Same-origin script with multiple sha256 hashes, including unknown algorithm.",
+        "matching-digest.js",
+        "sha256-EKclUXH9SRRUv3FmL7bIEV0z2s3EvzHFxzHKCnfHT/E= foo666-deadbeefdeadbeefdeadbeefdeadbeefdeadbeefdead"
+    ).execute();
+
+    // TODO may need to change these to hash-source prefix, e.g. 'sha256' instead of 'SHA-256'
+
+    test(function() {assert_equals(getPrioritizedHashFunction('SHA-512', 'SHA-256'), 'SHA-512')},
```

Comment:
confused.. my reading of the spec doesn't say that getprioritizedhashfunction is an actual function available to pages. What are you calling? I don't see it defined either.